scripts/gen_code_completion.py

```diff
@@ -0,0 +1,87 @@
+# Python script to generate code completion sources for use with
+# https://github.com/hrsh7th/nvim-cmp. Outputs lua code.
+
+# Copyright (C) 2022    Lucas Ritzdorf, Luca Zeuch
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+import re
+import os
+from io import StringIO
+
+class StringBuilder:
+    _file_str = None
+
+    def __init__(self):
+        self.file_str = StringIO()
+
+    def Add(self, str):
+        self.file_str.write(str)
+
+    def __str__(self):
+        return self.file_str.getvalue()
+
+def gen_keyword_list(file) -> list:
```

Comment:
Instead of using a list, you can use a generator instead, which *generates* items lazily on demand (hence the name.)